### ISSUE TITLE
Add data source for transit gateway's Direct Connect gateway attachment

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#8590]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When a Direct Connect gateway is associated with a transit gateway a transit gateway attachment with resource type `direct-connect-gateway` is created.
We need a new data source to be able to obtain the ID of this attachment.
See [this comment](https://github.com/terraform-providers/terraform-provider-aws/issues/8490#issuecomment-490960986).

Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/8490
* https://github.com/terraform-providers/terraform-provider-aws/pull/8528

Possible syntax:

```hcl
data "aws_ec2_transit_gateway_dx_gateway_attachment" "example" {
  transit_gateway_id = "${aws_ec2_transit_gateway.example.id}"
  dx_gateway_id      = "${aws_dx_gateway.example.id}"
}
```

**Comment by bflad** [comment]
The new `aws_ec2_transit_gateway_dx_gateway_attachment` data source has been merged and will be released shortly with version 2.14.0 of the Terraform AWS Provider. Thanks, @ewbankkit!

**Comment by bflad** [comment]
This has been released in [version 2.14.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2140-june-06-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
